bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -159,34 +159,111 @@ protected State getNAThingProperty(String channelId) {
+        }
+
+        List<NAWelcomeSubEvent> subEvents = event.getEventList();
+        for(NAWelcomeSubEvent subEvent: subEvents) {
+            String detectedObjectType = translateDetectedObjectType(subEvent.getType());
+            detectedObjectTypes.add(detectedObjectType);
+        }
+        return detectedObjectTypes;
+    }
+
+    private static String translateDetectedObjectType(NAWelcomeSubEvent.TypeEnum type) {
+        switch (type) {
+            case HUMAN: return CAMERA_EVENT_OPTION_HUMAN_DETECTED;
+            case ANIMAL: return CAMERA_EVENT_OPTION_ANIMAL_DETECTED;
+            case VEHICLE: return CAMERA_EVENT_OPTION_VEHICLE_DETECTED;
+            default: throw new IllegalArgumentException("Unknown detected object type!");
+        }
+    }
+
+    private @Nullable String findEventMessage() {
+        if(lastEvent != null) {
+            @Nullable String message = lastEvent.getMessage();
+            if(message == null) {
+                @Nullable NAWelcomeSubEvent firstSubEvent = findFirstSubEvent(lastEvent);
+                if(firstSubEvent != null) {
+                    message = firstSubEvent.getMessage();
+                }
+            }
+            return message;
+        }
+        return null;
+    }
+
     /**
      * Returns the Url of the picture
      *
      * @return Url of the picture or null
      */
-    protected String getSnapshotURL(NAWelcomeSnapshot snapshot) {
-        String result = null;
-
-        if (snapshot != null && snapshot.getId() != null && snapshot.getKey() != null) {
-            result = WELCOME_PICTURE_URL + "?" + WELCOME_PICTURE_IMAGEID + "=" + snapshot.getId() + "&"
-                    + WELCOME_PICTURE_KEY + "=" + snapshot.getKey();
-        } else {
-            logger.debug("Unable to build snapshot url for Home : {}", getId());
+    protected @Nullable String findSnapshotURL() {
+        if(lastEvent != null) {
+            NAWelcomeSnapshot snapshot = lastEvent.getSnapshot();
+            if (snapshot == null) {
+                @Nullable NAWelcomeSubEvent firstSubEvent = findFirstSubEvent(lastEvent);
+                if (firstSubEvent != null) {
+                    snapshot = firstSubEvent.getSnapshot();
+                }
+            }
+
+            if (snapshot != null && snapshot.getId() != null && snapshot.getKey() != null) {
+                return WELCOME_PICTURE_URL + "?" + WELCOME_PICTURE_IMAGEID + "=" + snapshot.getId() + "&"
+                        + WELCOME_PICTURE_KEY + "=" + snapshot.getKey();
+            } else {
+                logger.debug("Unable to build snapshot url for Home : {}", getId());
+            }
         }
-        return result;
+        return null;
     }
 
     @Override
     protected @Nullable Integer getDataTimestamp() {
         return dataTimeStamp;
     }
+
+    private static @Nullable NAWelcomeSubEvent findFirstSubEvent(@Nullable NAWelcomeEvent event) {
```

Comment:
Based on the logic that calls this method, this parameter shouldn't be nullable.
```suggestion
    private static @Nullable NAWelcomeSubEvent findFirstSubEvent(NAWelcomeEvent event) {
```

Reply:
This is what a said: "... but I think the "feature" is also buggy. See method NAWelcomeHomeHandler#findFirstSubEvent(...). I had to make this method to be NULL-safe, but it is never called with NULL...". IntelliJ gives a warning because the member "lastEvent" is marked as nullable... That the variable is checked against NULL before isn't recognized...